### PR TITLE
(#1711) Normalize v4 listings to be viewed in v5 clients

### PR DESCRIPTION
**Pull request by hoffmabc**
Ok so this code should normalize the v4 contracts pricing components (listing price, shipping options) into the v5 format for viewing in client.

### CODE REVIEW

**Comment by cpacia** [comment]
I think this would be better to use `UnmarshalJSONSignedListing` which should convert the whole thing to v5.

File: api/jsonapi.go [hunk 1]
```diff
@@ -1546,11 +1546,36 @@ func (i *jsonAPIHandler) GETListing(w http.ResponseWriter, r *http.Request) {
...
+			Divisibility: uint32(currency.Divisibility),
+		}
+
+		sl.Listing.Item.BigPrice = strconv.FormatUint(sl.Listing.Item.Price, 10)
+
+		for idx, shippingOption := range sl.Listing.ShippingOptions {
+			for idx2, service := range shippingOption.Services {
+				service.BigPrice = strconv.FormatUint(service.Price, 10)
+				service.BigAdditionalItemPrice = strconv.FormatUint(service.AdditionalItemPrice, 10)
+				sl.Listing.ShippingOptions[idx].Services[idx2] = service
```
Comment (gubatron) [inline api/jsonapi.go]: (testing multiline git comments)

Here's a stupid (unnecessary) memory optimization, in terms of moving less bytes around, just updating the fields you want to in place on the array, not overwriting the entire object:
```golang
sl.Listing.ShippingOptions[idx].Services[idx2].BigPrice = strconv.FormatUint(service.Price, 10)
sl.Listing.ShippingOptions[idx].Services[idx2].BigAdditionalItemPrice = strconv.FormatUint(service.AdditionalItemPrice, 10)
```

or to make it prettier (3 lines but shorter):
```golang
serviceRef = &sl.Listing.ShippingOptions[idx].Services[idx2]
serviceRef.BigPrice = strconv.FormatUint(service.Price, 10)
serviceRef.BigAdditionalItemPrice = strconv.FormatUint(service.AdditionalItemPrice, 10)
```

Cheers

**Comment by hoffmabc** [comment]
Needs #1906 to be a predecessor here and potentially will render it unnecessary.

**Comment by placer14** [comment]
We decided to close this until #1906 is resolved. At that point we can decide if this needs to be reconsidered.

**Pull request closed without merging**